state-sync/state-sync-v2/state-sync-driver/src/continuous_syncer.rs

```diff
@@ -101,11 +101,18 @@ impl<
         // Fetch the highest epoch state (in storage)
         let highest_epoch_state = utils::fetch_latest_epoch_state(self.storage.clone())?;
 
-        // Initialize a new active data stream
+        // Fetch the consensus sync request target (if there is one)
         let sync_request_target = consensus_sync_request
             .lock()
             .as_ref()
             .map(|sync_request| sync_request.get_sync_target());
+        if sync_request_target.is_some() {
```

Comment:
it's not very obvious this will only be called once

Reply:
This is actually called every time we create a new stream when there's a pending consensus sync request (e.g., if the previous stream failed or there were malicious responses). But, it's okay because we're guaranteed the execute-commit queue is empty and the database is up-to-date.